### ISSUE TITLE
protobuf dependency rebuild broken post protobuf reshuffle

### ISSUE THREAD

**Issue by mhaberler** [machinekit/machinekit#998]
@strahlex  - each make run will rebuild the pb deps although they are supposed to be built only once - strong hint at a wrong pathname with the .d files and maybe not effective at all:

`````
mah@jessie64:~/machinekit-patch/src$ make -j4 OPT=-O0
Reading 324/325 dependency files
Done reading dependencies

Reading 0/0 realtime dependency files
Done reading realtime dependencies
make: Entering directory '/home/mah/machinekit-patch/src'
protoc create dependencies for machinetalk/proto/src/machinetalk/protobuf/canon.proto
protoc create dependencies for machinetalk/proto/src/machinetalk/protobuf/status.proto
protoc create dependencies for machinetalk/proto/src/machinetalk/protobuf/object.proto
protoc create dependencies for machinetalk/proto/src/machinetalk/protobuf/firmware.proto
protoc create dependencies for machinetalk/proto/src/machinetalk/protobuf/test.proto
protoc create dependencies for machinetalk/proto/src/machinetalk/protobuf/preview.proto
protoc create dependencies for machinetalk/proto/src/machinetalk/protobuf/rtapicommand.proto
protoc create dependencies for machinetalk/proto/src/machinetalk/protobuf/motcmds.proto
protoc create dependencies for machinetalk/proto/src/machinetalk/protobuf/task.proto
protoc create dependencies for machinetalk/proto/src/machinetalk/protobuf/emcclass.proto
protoc create dependencies for machinetalk/proto/src/machinetalk/protobuf/config.proto
protoc create dependencies for machinetalk/proto/src/machinetalk/protobuf/nanopb.proto
protoc create dependencies for machinetalk/proto/src/machinetalk/protobuf/value.proto
protoc create dependencies for machinetalk/proto/src/machinetalk/protobuf/log.proto
protoc create dependencies for machinetalk/proto/src/machinetalk/protobuf/rtapi_message.proto
protoc create dependencies for machinetalk/proto/src/machinetalk/protobuf/types.proto
protoc create dependencies for machinetalk/proto/src/machinetalk/protobuf/message.proto

`````

**Comment by mhaberler** [comment]
part of the problem:

the introduction of the machinetalk/protobuf package namespace causes protoc-gen-depends to save the dependency file with the machinetalk/protobuf/ prefix which is likely the cause of the mismatch, hence repeated re-generation attempts

so for example 
`````
/usr/bin/protoc \
--plugin=protoc-gen-depends=machinetalk/proto/scripts/protoc-gen-depends \
--proto_path=machinetalk/proto/src/ \
--proto_path=/usr/include \
--depends_out=" --cgen=machinetalk/build --pygen=lib/python/machinetalk/protobuf --jsgen=machinetalk/tutorial/htdocs/js/machinetalk --vpath=machinetalk/proto/src --vpath=/usr/include --vpath=/usr/include/google/protobuf/compiler":/tmp/deps \
 machinetalk/proto/src/machinetalk/protobuf/canon.proto

# generates
mah@jessie64:~/machinekit-patch/src$ ls -l /tmp/deps/machinetalk/protobuf/canon.d 
-rw-r--r-- 1 mah mah 1866 Jul 15 18:12 /tmp/deps/machinetalk/protobuf/canon.d
`````
this used to be `/tmp/deps/canon.d` pre-change

this is independent of the install target problem https://github.com/machinekit/machinekit/issues/999

**Comment by mhaberler** [comment]
fixed with #1004
